packages/rollup-plugin/src/rollupUtils.ts

```diff
@@ -6,30 +6,53 @@ import type { SourcemapInfo } from './types'
 /**
  * Extracts the data we need for sourcemap upload from the bundle
  */
-export function extractSourcemapDataFromBundle ( 
-  outputOptions: NormalizedOutputOptions, 
-  bundle: OutputBundle
+export function extractSourcemapDataFromBundle (
+  outputOptions: NormalizedOutputOptions,
+  bundle: OutputBundle,
+  ignorePaths: Array<string | RegExp> = []
 ): SourcemapInfo[] {
   const sourceMaps = Object.values(bundle).filter(isSourcemap)
-  
+
   return sourceMaps.map(sourcemap => {
     return formatSourcemapData(outputOptions, sourcemap)
-  })
+  }).filter((sourcemap) => isNotIgnored(sourcemap, ignorePaths))
 }
 
 function isSourcemap(file: OutputAsset | OutputChunk): file is OutputAsset {
-  return file.type === 'asset' && file.fileName.endsWith('.js.map')
+  if (file.type !== 'asset' || !file.fileName.endsWith('.js.map') || !file.source) {
+    return false
+  }
+
+  const source = typeof file.source === 'string' ? file.source : file.source.toString()
+  const json = JSON.parse(source)
+
+  return json.sourcesContent && json.sourcesContent.length > 0
+}
+
+function isNotIgnored(sourceMapInfo: SourcemapInfo, ignorePaths: Array<string | RegExp>) {
+  for (const ignorePath of ignorePaths) {
+    let regex = ignorePath
+    if (!(regex instanceof RegExp)) {
+      regex = new RegExp(regex)
```

Comment:
I'm wondering if treating strings as RegExp here could be confusing to the user. I wouldn't necessarily expect that passing `'foo.js'` would ignore `subfolder1/foo.js` and `subfolder2/foo.js`. I think I'd expect to have to pass a wildcard like `*/foo.js`. Here if you did pass in a wildcard like `'*/foo.js'` or `subfolder/*/foo`, we'd accept it but I don't think it would work. 

For example 
```
> let r = new RegExp('subfolder/*/foo.js')
undefined
> r.test('subfolder/bar/foo.js')
false
> 
```
This could be cleared up with documentation. I'm just thinking that using a wildcard for path matching rather than a regex has become pretty common, so people might expect it?

Reply:
That's a great suggestion! I switched to [picomatch](https://github.com/micromatch/picomatch) for glob pattern matching instead!